first_hour_on_vwb/working_with_resources.ipynb

```diff
@@ -855,58 +855,49 @@
     "        self.input_name = wu.TextInputWidget(\"<NAME>\", \"Name:\").get()\n",
     "        self.output = widgets.Output()\n",
     "        self.button = wu.StyledButton('Create dataset','Click to create a BigQuery dataset as a controlled resource.','plus').get()\n",
-    "        self.vb = widgets.VBox([\n",
-    "            self.label,\n",
-    "            self.warning,\n",
-    "            self.access_drop_down,\n",
-    "            self.cloning_drop_down,\n",
-    "            self.input_dataset_id,\n",
-    "            self.input_description,\n",
-    "            self.input_name,\n",
-    "            self.button,\n",
-    "            self.output\n",
-    "        ], layout=wu.vbox_layout)\n",
+    "        self.toggle_optional = wu.ShowOptionalCheckbox().get()\n",
+    "        self.toggle_optional.observe(self.toggle)\n",
+    "        self.required_fields = [\n",
+    "                self.toggle_optional, self.label, self.warning,\n",
+    "                self.access_drop_down, self.cloning_drop_down, self.input_description,\n",
+    "                self.input_name, self.button, self.output\n",
+    "            ]\n",
+    "        self.vb = widgets.VBox(children = self.required_fields,\n",
+    "            layout = wu.vbox_layout\n",
+    "        )\n",
     "        self.button.on_click(self.create_bq_dataset)\n",
     "\n",
+    "    def toggle(self,event):\n",
+    "        self.output.clear_output()\n",
+    "        with self.output:\n",
+    "            if self.toggle_optional.value == True:\n",
+    "                parameterList = []\n",
+    "                for r in self.required_fields:\n",
+    "                    parameterList.append(r)\n",
+    "                parameterList.insert(5,self.input_dataset_id)\n",
```

Comment:
The 5 here is a magic number. We at should at least have a comment explaining it. Something like "Inserting parameters at specific locations in the UI".